tests/integration/conftest.py

```diff
@@ -201,29 +313,43 @@ async def app_openstack_runner_fixture(
     http_proxy: str,
     https_proxy: str,
     no_proxy: str,
-    openstack_clouds_yaml: str,
+    clouds_yaml_contents: str,
+    network_name: str,
+    flavor_name: str,
+    existing_app: Optional[str],
 ) -> AsyncIterator[Application]:
     """Application launching VMs and no runners."""
-    application = await deploy_github_runner_charm(
-        model=model,
-        charm_file=charm_file,
-        app_name=app_name,
-        path=path,
-        token=token,
-        runner_storage="juju-storage",
-        http_proxy=http_proxy,
-        https_proxy=https_proxy,
-        no_proxy=no_proxy,
-        reconcile_interval=60,
-        constraints={
-            "root-disk": 20 * 1024,
-            "cores": 4,
-            "mem": 16 * 1024,
-            "virt-type": "virtual-machine",
-        },
-        config={OPENSTACK_CLOUDS_YAML_CONFIG_NAME: openstack_clouds_yaml},
-        wait_idle=False,
-    )
+    if existing_app:
+        application = model.applications[existing_app]
+    else:
+        application = await deploy_github_runner_charm(
+            model=model,
+            charm_file=charm_file,
+            app_name=app_name,
+            path=path,
+            token=token,
+            runner_storage="juju-storage",
+            http_proxy=http_proxy,
+            https_proxy=https_proxy,
+            no_proxy=no_proxy,
+            reconcile_interval=60,
+            constraints={
+                "root-disk": 50 * 1024,
+                "mem": 16 * 1024,
```

Comment:
What would be the use case of this?
I agree the charm might not need this much resource anymore, and we can refine this setting.
Is there a test that would require a different resource usage?

Reply:
Just out of curiosity, hardcoding does not look very nice because these values depend on a concrete infrastructure. I think these settings should match an existing openstack flavour. So every time we change the flavor, this config needs to be updated? I assume this is not a big deal as flavour changes should not happen often.